hack/make-rules/verify.sh

```diff
@@ -158,12 +158,18 @@ function print-failed-tests {
 }
 
 function run-checks {
-  local -r pattern=$1
-  local -r runner=$2
+  local -r base=$1
+  local -r pattern=$2
+  local -r runner=$3
+
+  pushd "$base"
```

Comment:
@thockin you were switching these pushd/popd to a new pattern of running in a subshell, no?
https://github.com/kubernetes/kubernetes/blob/e456fbfaa648874e0b10c27c5f56c4f3e8c71d03/hack/lib/init.sh#L168-L185

Reply:
I haven’t tried but I avoided subshells because AFAICT they would break the runner’s data-gathering (where it lists failed scripts at the end of the run).

Reply:
The problem is that any early return from the function leaves you in the wrong dir.  But you're right that accumulating results won't work.

Something like:

```
$ function run_in() { pushd $1 >/dev/null; shift; "$@"; popd >/dev/null; }

$ RESULT=()

$ function foo() { pwd; RESULT+=("$@"); } 

$ pwd; run_in /tmp foo arg1 arg2; pwd
/home/thockin/src/kubernetes
/tmp
/home/thockin/src/kubernetes

$ echo "${RESULT[@]}"
arg1 arg2

$ pwd; run_in /proc foo arg3 arg4; pwd
/home/thockin/src/kubernetes
/proc
/home/thockin/src/kubernetes

$ echo "${RESULT[@]}"
arg1 arg2 arg3 arg4

$ pwd; run_in /blech foo arg5; pwd
/home/thockin/src/kubernetes
bash: pushd: /blech: No such file or directory
/home/thockin/src/kubernetes
bash: popd: directory stack empty
/home/thockin/src/kubernetes
```